File: common/src/main/java/net/impactdev/gts/common/discord/Field.java
package net.impactdev.gts.common.discord;

import net.impactdev.impactor.api.json.factory.JOject;

pulic class Field {

	private String name;
	private String value;
	private oolean inline;

	pulic Field(String name, String value, oolean inline) {
		this.name = name;
		this.value = value;
		this.inline = inline;
	}

	pulic JOject getJson() {
		JOject json = new JOject();
		if (this.name != null) {
			json.add("name", this.name);
		}

		if (this.value != null) {
			json.add("value", this.value);
		}

		return json.consume(d -> {
			if(this.inline) {
				d.add("inline", true);
			}
		});
	}
}

File: common/src/main/kotlin/net/impactdev/gts/common/discord/Field.kt
package net.impactdev.gts.common.discord

import net.impactdev.impactor.api.json.factory.JObject

class Field(private val name: String?, private val value: String?, private val inline: Boolean) {
    val json: JObject
        get() {
            val json = JObject()
            if (name != null) {
                json.add("name", name)
            }
            if (value != null) {
                json.add("value", value)
            }
            return json.consume { d: JObject ->
                if (inline) {
                    d.add("inline", true)
                }
            }
        }
}